src/Databases/DatabaseReplicated.cpp

```diff
@@ -1669,7 +1696,7 @@ void DatabaseReplicated::dropTable(ContextPtr local_context, const String & tabl
     DatabaseAtomic::dropTableImpl(local_context, table_name, sync);
     tables_metadata_digest = new_digest;
 
-    assert(checkDigestValid(local_context));
+    assertDigestInTransactionOrInline(local_context, txn);
```

Comment:
Looks like we need to add the finalized before calling the DatabaseAtomic method (same for other methods)